addons/sourcemod/scripting/superzombiefortress.sp

```diff
@@ -20,113 +17,125 @@
 
 #define PLUGIN_VERSION "3.1.7"
 
-#define INT(%0)		view_as<int>(%0)
-
-#define MAX_DIGITS 	12 // 10 + \0 for IntToString. And negative signs.
-
-#define GOO_INCREASE_RATE		3
-
-Handle cookieFirstTimeSurvivor;
-Handle cookieFirstTimeZombie;
-Handle cookieNoMusicForPlayer;
-Handle cookieForceZombieStart;
-
-// Global State
-bool zf_bEnabled;
-bool zf_bNewRound;
-int zf_lastSurvivor;
-
-float zf_spawnSurvivorsLastDeath = 0.0;
-int zf_spawnSurvivorsKilledCounter;
-int zf_spawnZombiesKilledCounter;
-int zf_spawnZombiesKilledSpree;
-int zf_spawnZombiesKilledSurvivor[MAXPLAYERS+1] = 0;
-
-// Client State
-int zf_survivorMorale[MAXPLAYERS+1];
-int zf_hordeBonus[MAXPLAYERS+1];
-int zf_CapturingPoint[MAXPLAYERS+1];
-int zf_rageTimer[MAXPLAYERS+1];
-bool zf_screamerNearby[MAXPLAYERS+1] = false;
-
-bool g_bStartedAsZombie[MAXPLAYERS+1];
-float g_flStopChatSpam[MAXPLAYERS+1] = 0.0;
-bool g_bWaitingForTeamSwitch[MAXPLAYERS+1] = false;
-
-int g_iSprite; // Smoker beam
-
-// Global Timer Handles
-Handle zf_tMain;
-Handle zf_tMoraleDecay;
-Handle zf_tMainFast;
-Handle zf_tMainSlow;
-Handle zf_tHoarde;
-Handle zf_tDataCollect;
-Handle zf_tTimeProgress;
-
-// Cvar Handles
-Handle zf_cvForceOn;
-Handle zf_cvRatio;
-Handle zf_cvSwapOnPayload;
-Handle zf_cvSwapOnAttdef;
-Handle zf_cvTankHealth;
-Handle zf_cvTankHealthMin;
-Handle zf_cvTankHealthMax;
-Handle zf_cvTankTime;
-Handle zf_cvFrenzyChance;
-Handle zf_cvFrenzyTankChance;
-
-float g_fZombieDamageScale = 1.0;
-
-//int g_StartTime = 0;
-//int g_AdditionalTime = 0;
-
-ArrayList g_aFastRespawnArray;
-
-bool g_bBackstabbed[MAXPLAYERS+1] = false;
+#define TF_MAXPLAYERS		32
```

Comment:
I personally recommend 34 just in case of TVs, Replays, etc.
Also why not have this 1 more instead of having the `TF_MAXPLAYERS+1`?

Reply:
I still need to see myself how replays interacts with SourceMod/max client index, whenever if replays can be complete ignored, is valid or not in `IsValidClient` stock.

I made `TF_MAXPLAYERS` 32 because well, TF2 indeed support max 32 players, just world takes up 0 slot.  If i were to consider replays (33), may well then also consider 0/world (34).

Reply:
After some read, i see replay bot can be anywhere between client 1 to 33, so yes that would break and go outside of array. Should also update `IsValidClient` to ignore replays while at it.